src/jit/optimizer.cpp

```diff
@@ -993,8 +993,7 @@ bool Compiler::optIsLoopTestEvalIntoTemp(GenTreePtr testStmt, GenTreePtr* newTes
     // Make sure we have jtrue (vtmp != 0)
     if ((relop->OperGet() == GT_NE) && 
         (opr1->OperGet() == GT_LCL_VAR) &&
-        (opr2->OperGet() == GT_CNS_INT) && 
-        (opr2->gtIntCon.gtIconVal == 0))
+        opr2->IsZero())
```

Comment:
This would permit long and floating const zero, which wasn't permitted earlier.  Can it lead to correctness issues due to liberal condition being checked here?